finetune/base.py

```diff
@@ -837,42 +887,39 @@ def finetune_grid_search_cv(
         return max(aggregated_results, key=lambda x: x[1])[0]
 
     def process_long_sequence(self, zipped_data):
-        arr_encoded = [
-            self.input_pipeline._text_to_ids(d["X"]) for d in zipped_data
-        ]
-
-        flat_array_encoded = []
-        sequence_id = []
-        for i, ae in enumerate(arr_encoded):
-            for sample in ae:
-                flat_array_encoded.append(sample)
-                sequence_id.append(i)
-
         labels, batch_probas = [], []
-        for pred in self._inference(
-                zipped_data,
-                predict_keys=[PredictMode.PROBAS, PredictMode.NORMAL],
-                chunked_length=len(flat_array_encoded),
+        for chunk_idx, (
+            pred,
+            (sequence_id, arr_enc, start_of_doc, end_of_doc),
+        ) in enumerate(
```

Comment:
Could you add a comment here explaining what's happening in the zipped loop? Is the purpose just to return token index information along with the predictions from `_inference()`?